src/components/button-icon/button-icon.component.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Events from '../../utils/helpers/events';
+import Icon from '../icon/icon.component';
+import StyledButtonIcon from './button-icon.style';
+
+const ButtonIcon = ({ onAction, type, ...rest }) => {
+  const onKeyDown = (e) => {
+    if (Events.isEnterKey(e) || Events.isSpaceKey(e) || Events.isEscKey(e)) {
+      e.preventDefault();
+      onAction(e);
+    } else {
+      e.stopPropagation();
+    }
+  };
+
+  const buttonRef = React.createRef();
+
+  const handleOnAction = (e) => {
+    e.preventDefault();
+    onAction(e);
+  };
+
+  return (
+    <StyledButtonIcon
+      { ...rest }
+      onKeyDown={ onKeyDown }
+      onClick={ handleOnAction }
+    >
+      <Icon
+        type={ type }
+        ref={ buttonRef }
+      />
+    </StyledButtonIcon>
+  );
+};
+
+ButtonIcon.defaultProps = {
+  type: 'close'
+};
+
+ButtonIcon.propTypes = {
+  /** Icon type */
+  type: PropTypes.string,
```

Comment:
I think that the type should be required and `close` should be not set as default

Reply:
agree! Type added in: 55e0442
updated respective components to pass in `type='close'`
a844a07
e1581ba
f4eca6b
df4b990
650287d